compatibility with MXE

starius opened the pull request:
Source: https://github.com/mxe/mxe/blob/master/plugins/apps/qbittorrent-1-fixes.patch
(Change of winconf.pri was improved, commit "convert includes like <Windows.h> to lowercase" is not needed.)

naikel commented:
This is NOT how it's supposed to be done.

First there's a PR for that already.  Second you don't edit the configure file, since that is autogenerated.  You modify the qbittorrent.m4 file. 

Second for the libraries: you just can't ditch them.  That's crazy.  It should be something like this:
```diff
diff --git a/winconf-mingw.pri b/winconf-mingw.pri
index 0283d23..465e4cc 100644
--- a/winconf-mingw.pri
+++ b/winconf-mingw.pri
@@ -22,17 +22,17 @@ RC_FILE = qbittorrent_mingw.rc
 
 # Adapt the lib names/versions accordingly
 CONFIG(debug, debug|release) {
-  LIBS += libtorrent \
-          libboost_system-mgw45-mt-d-1_47 \
-          libboost_filesystem-mgw45-mt-d-1_47 \
-          libboost_thread-mgw45-mt-d-1_47
+  LIBS += libtorrent-rasterbar \
+          libboost_system-mt \
+          libboost_filesystem-mt \
+          libboost_thread_win32-mt
 } else {
-  LIBS += libtorrent \
-          libboost_system-mgw45-mt-1_47 \
-          libboost_filesystem-mgw45-mt-1_47 \
-          libboost_thread-mgw45-mt-1_47
+  LIBS += libtorrent-rasterbar \
+          libboost_system-mt \
+          libboost_filesystem-mt \
+          libboost_thread_win32-mt
 }
 
 LIBS += libadvapi32 libshell32 libuser32
-LIBS += libcrypto.dll libssl.dll libwsock32 libws2_32 libz libiconv.dll
+LIBS += libcrypto libssl libwsock32 libws2_32 libz libiconv
 LIBS += libpowrprof
```

Just removing the file versions it works with MXE.

starius commented:
> First there's a PR for that already.

It is https://github.com/qbittorrent/qBittorrent/pull/4631. I have excluded commit 0365bd1c17aa2bfcc6f264864e6804746cf83d45 (do not check qmake existance).

> Second for the libraries: you just can't ditch them. That's crazy. It should be something like this

Fixed.

Note that I removed hardcoded paths to libs like "C:/qBittorrent/boost_1_51_0/stage/lib".

I added `QMAKE_LRELEASE` to `QT_QMAKE` command in both `configure` and `configure.ac`. However when I run `autoconf` I get a lot of differences between old `configure` and new `configure`. Probably my version of `autoconf` is different.

sledgehammer999 commented:
I don't understand how configure works in this case. It generates a `conf.pri` file which is only included by `unixconf.pri` and `macxconf.pri`. It isn't included by the `winconf*.pri` files at all.

Also the changes in winconf.pri aren't desired. They serve as a guide (on what to edit) for the newcomers that want to build on windows.

sledgehammer999 commented:
>Also the changes in winconf.pri aren't desired. They serve as a guide (on what to edit) for the newcomers that want to build on windows.

Maybe comment the paths instead? @qbittorrent/qbittorrent-frequent-contributors is that ok?

starius commented:
> Also the changes in winconf.pri aren't desired. They serve as a guide (on what to edit) for the newcomers that want to build on windows.

@sledgehammer999, I have excluded commit 62ec26a307a7502ea8aadb6bca0b4ff445e60810, it is not needed for MXE.

starius commented:
> it is not needed for MXE.

Oops, it is needed:

```
linking release/qbittorrent.exe
/bin/sh: 1: cannot open path-according-to-the-build-options-chosen: No such file
```

I have commented paths out.

starius commented:
> I don't understand how configure works in this case.

Neither do I. I changed it to pass QMAKE_LRELEASE to qmake as a environment variable. Maybe there is a better way to do it.

Chocobo1 commented:
>Maybe comment the paths instead? @qbittorrent/qbittorrent-frequent-contributors is that ok?

IMO it's better to move these to `winconf-msvc.pri` instead of comment out.

starius commented:
> IMO it's better to move these to winconf-msvc.pri instead of comment out.

Are you sure that nobody uses them with MinGW?

By the way, only one path (`C:/qBittorrent/RC_0_16/bin/<path-according-to-the-build-options-chosen>`) build compilation errors, because of "<" and ">". I have updated the pull request removing "<" and ">" from the path.

sledgehammer999 commented:
>IMO it's better to move these to winconf-msvc.pri instead of comment out.

Nope. They can be used when building with mingw too.

Chocobo1 commented:
>Nope. They can be used when building with mingw too.

Well never mind (I think we are not talking about the same thing) and the commits are different again.

starius commented:
Current version only changes `C:/qBittorrent/RC_0_16/bin/<path-according-to-the-build-options-chosen>` to `C:/qBittorrent/RC_0_16/bin/path-according-to-the-build-options-chosen`. Removing or commenting of the section is not needed.

starius commented:
Please review this Pull Request.

sledgehammer999 commented:
I'll try to do it tomorrow.

sledgehammer999 commented on winconf.pri:
```diff
@@ -53,7 +53,7 @@ CONFIG(debug, debug|release) {
 CONFIG += strace_win
 
-win32-g++ {
+*-g++* {
```
Will it work if you have `win32-g++*` instead?

starius replied:
It is not documented. According to [SO answers](http://stackoverflow.com/questions/14523148), it will.
I don't know who to get or set this string in qmake.

sledgehammer999 replied:
The original string is `win32-g++` and you change it to `*-g++*`. I am talking about changing it to `win32-g++*` which means any g++ on Windows.
The question is will `win32-g++*` work for MXE? Can you test it? If it does work then change it to what I suggested.

starius replied:
`win32-g++*` works in MXE. The PR was updated.

sledgehammer999 commented:
Use `git rebase -i HEAD~6` to make the following changes. When you use that command a txt file will be opened with your editor. Read the comments in that file to understand what is happening.

1. Your last commit is actually overwriting your first. So put the last's commit line above the first. Then in the first commit (now second) change the verb to `fixup`.
2. The commits `winconf-mingw.pri: remove versions from libraries` and `winconf-mingw.pri: remove ".dll" from libraries` do changes in the same file. The changes are very similar so you need just one commit. So `fixup` or `squash` the later commit.

After you're done `git push --force origin mxe-compat`. This PR will autoupdate and I can merge it. Leave a message so I get notified.

sledgehammer999 commented:
I am asking you this because I try to keep a clean and short commit history. (but not an overly compacted one)

starius commented:
Rebased. I haven't done it before to keep changes of the PR clear.

sledgehammer999 commented:
Thank you.
